src/com/palmergames/bukkit/towny/event/PlotPreChangeTypeEvent.java

```diff
@@ -0,0 +1,73 @@
+package com.palmergames.bukkit.towny.event;
+
+import com.palmergames.bukkit.towny.object.TownBlock;
+import com.palmergames.bukkit.towny.object.TownBlockType;
+import org.bukkit.Bukkit;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+public class PlotPreChangeTypeEvent extends Event implements Cancellable {
+    public static final HandlerList handlers = new HandlerList();
+    private TownBlockType oldType;
+    private TownBlockType newType;
+    private TownBlock townBlock;
+	private String cancelMessage = "Sorry this event was cancelled";
+	private boolean isCancelled = false;
+
+    @Override
+    public HandlerList getHandlers() {
+        return handlers;
+    }
+
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+
+    /**
+	 * Changes a plot's TownBlockType
+	 * 
+     * @param oldType- Old Type
+     * @param newType - New Type
+	 * @param townBlock - Plot to target
+     */
+    public PlotPreChangeTypeEvent(TownBlockType oldType, TownBlockType newType, TownBlock townBlock) {
+    	super(!Bukkit.getServer().isPrimaryThread());
+        this.newType = newType;
+        this.oldType = oldType;
+        this.townBlock = townBlock;
+    }
+
+    public TownBlockType getNewType() {
+        return newType;
+    }
+
+    public TownBlockType getOldType() {
+        if (oldType == null) {
+            return TownBlockType.WILDS; // Considering the further fact we know null is wilderness if there is no old type, it has to have been wilderness.
```

Comment:
The wilds plot type is not actually wilderness. TownBlocks that don't exist at all are wilderness. getOldType() will probably never return null as TownBlocks always have a type, even if it is the default RESIDENTIAL type.

Reply:
This could just return townBlock.getType().